src/components/Pressable/GenericPressable/PropTypes.js

```diff
@@ -4,6 +4,19 @@ import CONST from '../../../CONST';
 
 const stylePropTypeWithFunction = PropTypes.oneOfType([stylePropType, PropTypes.func]);
 
+/**
+ * Marks prop as required if the component is accessible
+ * @param {Object} props
+ * @param {String} propName
+ * @returns {Error} Error if prop is required
+ */
+const requireIfAccessible = (props, propName) => {
```

Comment:
Not sure how this is supposed to work. But still I don't see it working, since if i just make accessible true and not pass accessibilityLabel I don't get any error. Same goes the other way. 

But anyways I think this change is not related to this PR, if required we should bring in other changes related to this which should be in pipeline i suppose.

Reply:
Yeah, you're right, but It creates console error, as it is required. 
It should be in seperate PR – accessibilityLabel is required when PressableWithFeedback is set to be not accessible.

Reply:
I am not sure if i understand this comment. Are you saying we are getting console error without this change and we are adding this to fix the console error? Or we are adding this as an improvement to allow dynamic change of accessibility label?

Reply:
@robertKozik

Reply:
In other words, before any changes from this PR, proptype for accessibilityLabel looked like this:
```javascript
accessibilityLabel: propTypes.string.required,
```
so it will case error when no accessibilityLabel is passed.

Reply:
Got it. So the problem we are trying to solve is, `accessibilityLabel` is marked as `required` already, but `it is not actually required` if `accessible` is `false`. Since currently RadioButtonWithLabel doesn't pass accessibilityLabel it will start throwing error. So we are implementing the above code to make it dynamically check and throw console error in case if `accessible` value is `true`. 

But my concerns are `Ideally if a prop is not everytime required we shouldn't mark it as required`. But I think it's good to involve other reviewer and get their thoughts as well.

So, the questions and options are,

1) Are we fine the above code which dynamically marks `accessibilityLabel` as `required` based on the `accessible` prop given? (I am not sure if we have used similar code in other places as well in our codebase, and its fine to go this way instead of directly marking proptypes)

2) Or should we remove the `accessibilityLabel` from being `required`

3) Any other options we need to look for?

@dangrous

Reply:
Hm yeah that's an interesting situation. I think the correct behavior is indeed only requiring the `accessibilityLabel` if `accessible` is true. I see what you're saying @abdulrahuman5196 about not marking it as required if it's not really... but I'd err on the side of accessibility and use this dynamic check. That is, I'd rather have the code be a little unique than let a component that should have a label through without one. I found at least one other place where we dynamically require props: https://github.com/Expensify/App/blob/5fab2de66ee1666d75ef6bae4f56fd04f94eb6f6/src/components/CheckboxWithLabel.js#L36-L39 - which isn't quite the same, but it's not completely new.

If you're not convinced though, it's totally fine to open a discussion on Slack to see if others have thoughts!

Reply:
Got it. Wanted to confirm anyways on my concerns. I am good now.